pytensor/tensor/rewriting/uncanonicalize.py

```diff
@@ -31,33 +31,34 @@
 
 """
 
-from pytensor import scalar as ps
 from pytensor.graph.rewriting.basic import copy_stack_trace, node_rewriter
 from pytensor.tensor.basic import Alloc, alloc, constant
-from pytensor.tensor.elemwise import CAReduce, DimShuffle
-from pytensor.tensor.math import Argmax, Max, MaxAndArgmax, Min, neg
+from pytensor.tensor.elemwise import DimShuffle
+
+# from pytensor.tensor.math import Argmax, Max, MaxAndArgmax, Min, neg
+from pytensor.tensor.math import Min, TensorMax, neg
 from pytensor.tensor.rewriting.basic import register_uncanonicalize
 from pytensor.tensor.shape import Reshape, reshape
 from pytensor.tensor.subtensor import Subtensor
 
 
-@register_uncanonicalize
-@node_rewriter([MaxAndArgmax])
-def local_max_and_argmax(fgraph, node):
-    """
-    If we don't use the argmax, change it to a max only.
-    """
-    if isinstance(node.op, MaxAndArgmax):
-        axis = node.op.axis
-        if len(fgraph.clients[node.outputs[1]]) == 0:
-            new = Max(axis)(node.inputs[0])
-            copy_stack_trace(node.outputs[0], new)
-            return [new, None]
+# @register_uncanonicalize
+# @node_rewriter([MaxAndArgmax])
+# def local_max_and_argmax(fgraph, node):
+#     """
+#     If we don't use the argmax, change it to a max only.
+#     """
+#     if isinstance(node.op, MaxAndArgmax):
+#         axis = node.op.axis
+#         if len(fgraph.clients[node.outputs[1]]) == 0:
+#             new = Max(axis)(node.inputs[0])
+#             copy_stack_trace(node.outputs[0], new)
+#             return [new, None]
 
-        if len(fgraph.clients[node.outputs[0]]) == 0:
-            new = Argmax(axis)(node.inputs[0])
-            copy_stack_trace(node.outputs[0], new)
-            return [None, new]
+#         if len(fgraph.clients[node.outputs[0]]) == 0:
+#             new = Argmax(axis)(node.inputs[0])
+#             copy_stack_trace(node.outputs[0], new)
+#             return [None, new]
```

Comment:
This function seems irrelevant now. We have TensorMax only now so we do not need to drop argmax from the graph.

Reply:
Yup, the rewrite can be dropped